pymaker/dss.py

```diff
@@ -496,6 +294,18 @@ def frob(self, ilk: Ilk, urn_address: Address, dink: Wad, dart: Wad, collateral_
         return Transact(self, self.web3, self.abi, self.address, self._contract,
                         'frob', [ilk.toBytes(), urn_address.address, v.address, w.address, dink.value, dart.value])
 
+    def get_wipe_all_dart(self, ilk: Ilk, urn: Address) -> Wad:
+        """Returns the amount of Dai required to wipe an urn without leaving any dust
+        adapted from https://github.com/makerdao/dss-proxy-actions/blob/master/src/DssProxyActions.sol#L200"""
+        assert isinstance(urn, Address)
+        assert isinstance(ilk, Ilk)
+        assert ilk.rate >= Ray.from_number(1)
+
+        rad: Rad = Rad(self.urn(ilk, urn).art) * Rad(ilk.rate)
```

Comment:
I might just be confused here, but the types seem off.

`rad = wad * ray`, because `10^18 * 10^27= 10^45`

I see you're scaling that to:
`X = rad(wad) * rad(ray)`

But does this somehow prevent the magnitudes from scaling?
`10^45 * 10^45 = 10^90`

This code might be correct, but if it is, there's a `rad` divided out somewhere.

Reply:
With a few exceptions, pymaker's numeric classes do not perform arithmetic across type.  For example, a `Wad` times a `Ray` would produce a runtime error, not a `Rad`.  As such, we generally convert operands to the greatest common precision, and then cast to get the desired precision.